### PR TITLE
Remove model.AppError from searchlayer/post_layer.go

### LINKED ISSUE #16417: Remove model.AppError from searchlayer/post_layer.go
We are in the process of removing `model.AppError` completely from the `store` package. We have removed it from the `sqlstore` package, but there are still a couple left in some other packages under `store`.

This ticket is about removing `model.AppError` from searchlayer/post_layer.go

The way to do that would be to use the different error types in store/errors.go if a particular error falls in that category. And if it's just a generic error, then just return an `errors.Wrapf` wrapping the original error.

If you have questions about the ticket or you need any help, feel free to contact `agniva.de.sarker` in https://community.mattermost.com/

----------------------------------

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

### PR DESCRIPTION
**Pull request by msal4**
#### Summary

Removes all `model.AppError` references in searchlayer/post_layer.go.

#### Ticket Link

Fixes #16417
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/searchlayer/post_layer.go [hunk 1]
```diff
@@ -140,14 +137,7 @@ func (s SearchPostStore) searchPostsInTeamForUserByEngine(engine searchengine.Se
 	userChannels, nErr := s.rootStore.Channel().GetChannels(teamId, userId, paramsList[0].IncludeDeletedChannels, 0)
```
Comment (agnivade) [inline store/searchlayer/post_layer.go:137]: We can change the `nErr` to `err`

**Review by agnivade — COMMENTED** [review]
Just one small comment. Thanks @msal4 !

**Comment by agnivade** [comment]
@msal4 - We have a scoping issue, due to which the vet check is failing. Let's use `err2` for it.

**Comment by msal4** [comment]
@agnivade what if I declare `err` on the first line would it solve the issue? like so:
```go
var err error
if err = model.IsSearchParamsListValid(paramsList); err != nil {
	return nil, err
}
//...
```

**Comment by agnivade** [comment]
Sure, that works too. But it's more idiomatic to keep `err` variables in their own scopes to prevent shadowing issues in future.

**Comment by agnivade** [comment]
Hey @msal4 - just checking in on this one. Let me know if there is anything I can help with.

**Comment by msal4** [comment]
@agnivade sorry for the late commit, I made the necessary changes

**Review by agnivade — APPROVED** [review]
Thanks!

**Review by rvillablanca — APPROVED** [review]
LGTM

**Comment by agnivade** [comment]
This is an internal refactor and does not need QA testing.

**Pull request merged**